f/connectors/alerts/alerts_gcs.py

```diff
@@ -374,14 +365,29 @@ def prepare_alerts_metadata(alerts_metadata, territory_id):
 
     logger.info("Successfully prepared alerts metadata.")
 
-    # Generate alert statistics based on the latest row in the filtered DataFrame
-    latest_row = filtered_df.sort_values(
-        by=["year", "month"], ascending=[False, False]
-    ).iloc[0]
+    # Determine latest month and year
+    latest_month_year = (
+        filtered_df[["month", "year"]]
+        .drop_duplicates()
+        .sort_values(by=["year", "month"], ascending=[False, False])
+        .iloc[0]
+    )
+
+    # Filter for rows matching the latest month and year
+    latest_rows = filtered_df[
+        (filtered_df["month"] == latest_month_year["month"])
+        & (filtered_df["year"] == latest_month_year["year"])
+    ]
```

Comment:
When I saw this new code my mind went to "but can't you do it in a one-liner" and then noticed that's what the old code is.

I believe your point is that you need `latest_rows` to possibly be more than one row long.

You already stated that motivation in the PR description but please also mention it in an inline code comment.